src/main/java/uk/gov/hmcts/reform/bulkscan/payment/processor/config/SiteConfiguration.java

```diff
@@ -4,62 +4,46 @@
 import uk.gov.hmcts.reform.bulkscan.payment.processor.exception.SiteConfigurationException;
 
 import java.util.List;
-import java.util.Map;
-import java.util.stream.Collectors;
-import javax.annotation.PostConstruct;
+import java.util.Objects;
+import java.util.Optional;
 
 @ConfigurationProperties(prefix = "site-mappings")
 public class SiteConfiguration {
-    private List<Sites> sites;
+    private List<Site> sites;
 
-    private Map<String, String> poBoxToSiteIdMap;
-
-    public List<Sites> getSites() {
+    public List<Site> getSites() {
         return sites;
     }
 
-    public void setSites(List<Sites> sites) {
-        this.sites = sites;
-    }
-
-    @PostConstruct
-    private void mapPoBoxToSiteId() {
-        if (getSites() == null || getSites().isEmpty()) {
+    public void setSites(List<Site> sites) {
+        if (sites == null || sites.isEmpty()) {
             throw new SiteConfigurationException("Site configuration missing");
+        } else {
+            this.sites = sites;
         }
-        poBoxToSiteIdMap = getSites()
-            .stream()
-            .collect(Collectors.toMap(Sites::getPoBox, Sites::getSiteId));
     }
 
-    public String getSiteIdByPoBox(String poBox) {
-        return poBoxToSiteIdMap.get(poBox);
+    public Optional<String> getSiteIdByPoBox(String poBox) {
+        return sites
+            .stream()
```

Comment:
So every time requesting siteId for certain PO BOX we will do streaming? What was wrong by just storing key value map like we used?